packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -133,9 +133,10 @@ export class CfnResource extends CfnRefElement {
    * Ideally, use generated attribute accessors (e.g. `resource.arn`), but this can be used for future compatibility
    * in case there is no generated attribute.
    * @param attributeName The name of the attribute.
+   * @param sub optional, indicates if this is in a Fn::Sub call from cloudformation-include
    */
-  public getAtt(attributeName: string): Reference {
-    return CfnReference.for(this, attributeName);
+  public getAtt(attributeName: string, sub?: string): Reference {
+    return CfnReference.for(this, attributeName, sub);
```

Comment:
This I absolutely hate. Definitely a string here is not a great idea. Remember, this is a public method!

How about an enum `ReferenceContext` with 2 values, `PROPERTY_DEFINITION` and `FN_SUB_EXPRESSION`, and we say the default is `PROPERTY_DEFINITION`?

Reply:
Actually, I changed my mind. Leave this method unchanged.

Instead, just "inline" this expression in the only place that you're calling this with a `sub` parameter, which is in `cfn-parse.ts`.

Sorry for the churn, but I want to avoid changing any public APIs as part of this change (`CfnReference.for()` is different, as that's a private class of the `core` module).